daml-foundations/daml-ghc/daml-prim-src/GHC/Show.daml

```diff
@@ -21,97 +21,55 @@ import GHC.CString (fromString)
+--   parentheses. For example, if `d` is `0` then the result
 --   is never surrounded in parentheses; if `d` is `11` it is always
 --   surrounded in parentheses, unless it is an atomic expression.
 --
--- * If the constructor is defined using record syntax, then 'show'
+-- * If the constructor is defined using record syntax, then `show`
 --   will produce the record-syntax form, with the fields given in the
 --   same order as the original declaration.
---
--- For example, given the declarations
---
--- ```
--- infixr 5 :^:
--- data Tree a =  Leaf a  |  Tree a :^: Tree a
--- ```
---
--- the derived instance of 'Show' is equivalent to
---
--- ```
--- instance (Show a) => Show (Tree a) where
---   showsPrec d (Leaf m) = showParen (d > app_prec) $
---        showString "Leaf " . showsPrec (app_prec+1) m
---     where app_prec = 10
---
---   showsPrec d (u :^: v) = showParen (d > up_prec) $
---        showsPrec (up_prec+1) u .
---        showString " :^: "      .
---        showsPrec (up_prec+1) v
---     where up_prec = 5
--- ```
---
--- Note that right-associativity of `:^:` is ignored.  For example,
---
--- * `show (Leaf 1 :^: Leaf 2 :^: Leaf 3)` produces the string
---   `"Leaf 1 :^: (Leaf 2 :^: Leaf 3)"`.
 
 class  Show a  where
     {-# MINIMAL showsPrec | show #-}
 
-    -- | Convert a value to a readable `String`.
-    --
-    -- `showsPrec` should satisfy the law
-    --
-    -- ```
-    -- showsPrec d x r ++ s  ==  showsPrec d x (r ++ s)
-    -- ```
-    --
-    -- Derived instances of 'Text.Read.Read' and 'Show' satisfy the following:
-    --
-    -- * `(x,"")` is an element of
-    --   `(Text.Read.readsPrec d (showsPrec d x ""))`.
-    --
-    -- That is, `Text.Read.readsPrec` parses the string produced by
-    -- `showsPrec`, and delivers the value that `showsPrec` started with.
+    -- | Convert a value to a readable `Text` value. Unlike `show`,
+    -- showsPrec` should satisfy the rule `showsPrec d x r ++ s  ==
+    -- showsPrec d x (r ++ s)`
 
     showsPrec : Int    -- ^ the operator precedence of the enclosing
                         -- context (a number from `0` to `11`).
                         -- Function application has precedence `10`.
               -> a      -- ^ the value to be converted to a 'String'
               -> ShowS
 
-    -- | A specialised variant of 'showsPrec', using precedence context
-    -- zero, and returning an ordinary 'String'.
+    -- | Convert a value to a readable `Text` value.
     show      : a   -> Text
 
-    -- | The method 'showList' is provided to allow the programmer to
-    -- give a specialised way of showing lists of values.
-    -- For example, this is used by the predefined 'Show' instance of
-    -- the 'Char' type, where values of type 'String' should be shown
-    -- in double quotes, rather than between square brackets.
+    -- | Allows you to show lists of values.
+    -- For example, this is used by the `Char` type, where values of type
```

Comment:
We don't have Char

Reply:
@neil-da Shall we just drop `showList` from the class?